apis/src/pages/profile_view.rs

```diff
@@ -1,110 +1,261 @@
+                    <ToggleGroupRoot
+                        attr:class="flex"
+                        kind=ToggleGroupKind::Multiple {
+                            value: ToggleGroupMultiple::none().into(),
+                            default_value: ctx
+                                .speeds
+                                .get_untracked()
+                                .iter()
+                                .map(|s| s.to_string())
+                                .collect::<Vec<_>>()
+                                .into(),
+                            on_value_change: Some(
+                                Callback::from(move |value: Vec<String>| {
+                                    let value: Vec<_> = value
+                                        .into_iter()
+                                        .map(|s| GameSpeed::from_str(&s).unwrap())
+                                        .collect();
+                                    ctx.speeds.set(value.clone());
+                                    first_batch(
+                                        value,
+                                        ctx.tab_view.get(),
+                                        (username(), ctx.user_color.get()),
+                                    );
+                                }),
+                            ),
+                        }
+                    >
+
+                        <ToggleGroupItem value="Bullet" attr:class=toggle_classes>
+                            "Bullet"
+                        </ToggleGroupItem>
+                        <ToggleGroupItem value="Blitz" attr:class=toggle_classes>
+                            "Blitz"
+                        </ToggleGroupItem>
+                        <ToggleGroupItem value="Rapid" attr:class=toggle_classes>
+                            "Rapid"
+                        </ToggleGroupItem>
+                        <ToggleGroupItem value="Classic" attr:class=toggle_classes>
+                            "Classic"
+                        </ToggleGroupItem>
+                        <ToggleGroupItem value="Correspondence" attr:class=toggle_classes>
+                            "Correspondence"
+                        </ToggleGroupItem>
+                        <ToggleGroupItem value="Untimed" attr:class=toggle_classes>
+                            "Untimed"
+                        </ToggleGroupItem>
+                    </ToggleGroupRoot>
+                    {children()}
+                </div>
+            </Show>
+        </div>
+    }
+}
+
+#[component]
+pub fn DisplayGames(tab_view: ProfileGamesView) -> impl IntoView {
+    let ctx = expect_context::<ProfileGamesContext>();
+    ctx.tab_view.set(tab_view);
+    let username = move || ctx.user.get().map_or(String::new(), |user| user.username);
+    let el = create_node_ref::<html::Div>();
+    let ws = expect_context::<WebsocketContext>();
+    create_effect(move |_| {
```

Comment:
Same as the other create_effect.

Reply:
I will resolve this so I wont get lost with 2 discussion threads